### DISCUSSION TITLE
Proposal: Add Kotest to korlibs?

### DISCUSSION THREAD

**Discussion by soywiz** [korlibs/korge#1247] [Ideas and discussions]
Initial discussion: https://github.com/korlibs/korge/pull/1244

Also proposed here: https://github.com/korlibs/korge/pull/1185

**Comment by soywiz** [comment]
My own points here, please feel free to share more:

PROs:
* Modern API
* Multiplatform (supporting all the platforms we want to support)

CONs:
* Requires a IntelliJ plugin to run tests there
* Have to validate that it can show differences between outputs in IntelliJ

**Reply by ygdrasil-io** [reply]
I'm using it with my team to train on TDD with JVM target, the framework is working fine with Maven Tasks and From the IntelliJ Plugin.

But on multiplatform the plugin is not yet working as describe here :
https://kotest.io/docs/intellij/intellij-plugin.html 

I tried it on a personnal project : https://github.com/ygdrasil-io/Delaunator-Kt/tree/feature/refactoring (be sure to checkout the correct branch)
You can run them with `gradle allTests`, tests are running with JMV,JS and native host taget (OSX on my case).

The multiplatform experience is not so great without the plugin but the DSL help a lot to increase meaning of created tests. 
Still, this is not so hard to reproduce if we find a way to create tests dinamicaly on Kotlin test library.

**Reply by Kietyo** [reply]
So maybe another reason to just have something lightweight built into Korlibs itself?
https://github.com/korlibs/korge/pull/1244

I initially created the PR because this is something I built in my own project and thought it would be useful for others as well.
Also I would much prefer the truth syntax when writing tests in Korge rather than kotlin test syntax.
